openedx/core/djangoapps/content_libraries/serializers.py

```diff
@@ -46,6 +48,24 @@ class ContentLibraryMetadataSerializer(serializers.Serializer):
     has_unpublished_changes = serializers.BooleanField(read_only=True)
     has_unpublished_deletes = serializers.BooleanField(read_only=True)
     license = serializers.ChoiceField(choices=LICENSE_OPTIONS, default=ALL_RIGHTS_RESERVED)
+    can_edit_library = serializers.SerializerMethodField()
+
+    def get_can_edit_library(self, obj):
+        """
+        Verifies if the user in request has permission
+        to edit a library.
+        """
+        request = self.context.get('request', None)
+        if request is None:
+            return False
+        user = request.user
+        can_edit_library = False
+        library_obj = ContentLibrary.objects.get_by_key(obj.key)
+
+        if user:
+            can_edit_library = user.has_perm(permissions.CAN_EDIT_THIS_CONTENT_LIBRARY, obj=library_obj)
+
+        return can_edit_library
```

Comment:
not sure if it is relevant, but we can avoid the get call if we don't have a user here

```suggestion
        if not user:
            return False

        library_obj = ContentLibrary.objects.get_by_key(obj.key)
        return user.has_perm(permissions.CAN_EDIT_THIS_CONTENT_LIBRARY, obj=library_obj)
```

Reply:
Updated: https://github.com/openedx/edx-platform/pull/34934/commits/837727333b14bb66602a4047bf6edea7d89863aa#diff-abdb0ddf9b11dba593a3504d0f1a7acd76ccaf15e1be029efd16cb73c929984bR65